migration and address linkability at the server (#394)
quicwg/ops-drafts: draft-ietf-quic-applicability.md

Use of a non-zero-length connection ID for the server is strongly recommended if
any clients are behind a NAT or could be. A non-zero-length connection ID is

> also strongly recommended when active migration is supported. If a connection

Author: This section is introducing the CID and the fact that the CID changes on intended/active migration to a new path (e.g. in a handover for mobility) should be mentioned here as well from my point of view.

> is intentionally migrated to new path, an new connection ID is used to minimize

Reviewer: ```suggestion
is intentionally migrated to new path, a new connection ID is used to minimize
```

> linkability by network observers. The other QUIC endpoint uses the

Author: The CID is used to match connection at the receiver. A _new_ CID is used to avoid linkability on path.

> linkability by network observers. The other QUIC endpoint uses the

Reviewer 2: Right. Two concepts that this mixes together.

> linkability by network observers; of the other QUIC endpoint, however, is
> of course able to link different addresses to the same connection and entity
> if a non-zero-length connection ID is provided.

Reviewer: ```suggestion
linkability by network observers. The other QUIC endpoint uses the
connection ID to link different addresses to the same connection
and entity if a non-zero-length connection ID is provided.
```

using the connection ID.

While sufficiently robust connection ID generation schemes will mitigate
linkability for on-path observers, they do not provide full protection.

> Of course, the server uses the connection ID to associate packets to the same

Reviewer: ```suggestion
The server uses the connection ID to associate packets to the same
```

> Of course, the server uses the connection ID to associate packets to the same
> QUIC connection and as such can also link differeet IP addresses to the same
> user. Further, analysis of

Reviewer: ```suggestion
The server uses the connection ID to associate packets to a QUIC
connection and as such can also link different IP addresses to the same
user. Further, analysis of
```

considerations for the underlying transport protocol are relevant for
applications using QUIC, as well. Considerations on linkability, replay attacks,
and randomness discussed in {{!QUIC-TLS}} should be taken into account when

> deploying and using QUIC. Further, migration to an new address exposes

Reviewer 2: This new point is worth a new paragraph.

> Using migration reveals to the server that the same client controls multiple addresses.

Reviewer 3: ```suggestion
deploying and using QUIC.

Further, migration to an new address exposes
```

> a linkage between client addresses to the server which should also be

Reviewer: It may also be to the path, so I'd drop 'to the server'.

```suggestion
a linkage between client addresses which should also be
```

Author: I think I would rather prefer to be explicit about both cases here. 

```suggestion
a linkage between client addresses to the server and may expose this linkage
also to the path if the connection ID cannot be changed or flows can
otherwise be correlated. When migration is supported, this needs to be
```
